third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
+    #[semantics = r#"
+        stack >> val
+        stack >> vec_ref
+        (*vec_ref) << val
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> val_ty 
+        assert val_ty == elem_ty 
+        ty_stack >> ref_ty 
+        assert ref_ty == &mut vector<elem_ty>
+    "#]
     VecPushBack(SignatureIndex),
-    /// Pop an element from the end of vector. Aborts if the vector is empty.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., vector_reference -> ..., element```
+
+    #[group = "vector"]
+    #[description = r#"
+        Pop an element from the end of vector. 
+        Aborts if the vector is empty.
+    "#]
+    #[static_operands = "[elem_ty_idx]"]
+    #[semantics = r#"
+        stack >> vec_ref
+        (*vec_ref) >> val
+        stack << val
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> ref_ty 
+        assert ref_ty == &mut vector<elem_ty> 
+        ty_stack << val_ty
+    "#]
     VecPopBack(SignatureIndex),
-    /// Destroy the vector and unpack a statically known number of elements onto the stack. Aborts
-    /// if the vector does not have a length N.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., vec[e1, e2, ..., eN] -> ..., e1, e2, ..., eN```
+
+    #[group = "vector"]
+    #[description = r#"
+        Destroy the vector and unpack a statically known number of elements onto the stack. 
+        Abort if the vector does not have a length `n`.
+    "#]
+    #[static_operands = "[elem_ty_idx] [n]"]
+    #[semantics = r#"
+        stack >> vector[elem_1, ..., elem_n]
+        stack << elem_1
+        ...
+        stack << elem_n
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> ty 
+        assert ty == vector<elem_ty> 
+        ty_stack << [elem_ty]*n
+    "#]
     VecUnpack(SignatureIndex, u64),
-    /// Swaps the elements at two indices in the vector. Abort the execution if any of the indice
-    /// is out of bounds.
-    ///
-    /// ```..., vector_reference, u64_value(1), u64_value(2) -> ...```
+
+    #[group = "vector"]
+    #[description = r#"
+        Swaps the elements at two indices in the vector. 
+        Abort the execution if any of the indice is out of bounds.
+    "#]
+    #[static_operands = "[elem_ty_idx]"]
+    #[semantics = r#"
+        stack >> j
+        stack >> i
+        stack >> vec_ref
+        (*vec_ref)[i], (*vec_ref)[j] = (*vec_ref)[j], (*vec_ref)[i]
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> ty1
+        ty_stack >> ty2
+        ty_stack >> ty3
+        assert ty1 == u64
+        assert ty2 == u64
+        assert ty3 == &vector<elem_ty>
+    "#]
     VecSwap(SignatureIndex),
-    /// Push a U16 constant onto the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```... -> ..., u16_value```
+
+    #[group = "stack_and_local"]
+    #[description = "Push a u16 constant onto the stack."]
+    #[static_operands = "[u16_value]"]
+    #[semantics = "stack << u16_value"]
+    #[paranoid_post = "ty_stack << u16"]
     LdU16(u16),
```

Comment:
Weird that these show up down here instead of with the other `LdUx` above. Would be nice if they were together.

Reply:
The reason why these instructions appear at the end is because they were added at a later time. 

Now since we don't derive the serialization of the instructions it may very well be fine to change the order, but that's better done in a separate PR, with adequate verification.